lib/project-manager/src/main/scala/org/enso/projectmanager/infrastructure/repository/ProjectFileRepository.scala

```diff
@@ -122,4 +196,44 @@ class ProjectFileRepository[F[+_, +_]: Sync: ErrorChannel: CovariantFlatMap](
         StorageFailure(failure.toString)
       )
 
+  /** @inheritdoc **/
+  override def moveProjectToTargetDir(
+    projectId: UUID
+  ): F[ProjectRepositoryFailure, File] = {
+    getProject(projectId)
+      .flatMap { project =>
+        val targetPath = getTargetPath(project)
+        if (targetPath.toString == project.path.get) {
+          CovariantFlatMap[F].pure(targetPath)
+        } else {
+          moveProjectDir(project, targetPath) *>
+          updateProjectDir(projectId, targetPath) *>
+          CovariantFlatMap[F].pure(targetPath)
+        }
+      }
+
+  }
+
+  private def updateProjectDir(projectId: UUID, targetPath: File) = {
+    indexStorage
+      .modify { index =>
+        val updated = index.update(projectId)(
+          _.copy(path = Some(targetPath.toString))
+        )
+        (updated, ())
+      }
+      .mapError(_.fold(convertFileStorageFailure))
+  }
+
+  private def moveProjectDir(project: Project, targetPath: File) = {
+    fileSystem
+      .move(new File(project.path.get), targetPath)
```

Comment:
Make sure that there are no open files. Otherwise, it will fail on Windows.

Reply:
good point

Reply:
implemented